instrumentor/runtimemigration/runtimemigration.go

```diff
@@ -0,0 +1,314 @@
+	for namespace, workloadNames := range namespaces {
+		var daemonSets appsv1.DaemonSetList
+		err := clientset.List(ctx, &daemonSets, &client.ListOptions{Namespace: namespace})
+		if err != nil {
+			return fmt.Errorf("failed to list daemonsets in namespace %s: %v", namespace, err)
+		}
+
+		for _, ds := range daemonSets.Items {
+			// Checking if the daemonset is in the list of daemonsets that need to be processed
+			if contains(workloadNames, ds.Name) {
+
+				originalWorkloadEnvVar, _ := envoverwrite.NewOrigWorkloadEnvValues(ds.Annotations)
+				workloadInstrumentationConfigReference := workloadNames[ds.Name]
+				runtimeDetailsByContainer := workloadInstrumentationConfigReference.Status.RuntimeDetailsByContainer
+
+				for _, containerObject := range ds.Spec.Template.Spec.Containers {
+					err := handleContainerRuntimeDetailsUpdate(
+						containerObject,
+						*originalWorkloadEnvVar,
+						&runtimeDetailsByContainer)
+					if err != nil {
+						return fmt.Errorf("failed to process container %s in daemonset %s: %v", containerObject.Name, ds.Name, err)
+					}
+				}
+
+				// Update runtimeDetailsByContainer in workloadInstrumentationConfigReference
+				workloadInstrumentationConfigReference.Status.RuntimeDetailsByContainer = runtimeDetailsByContainer
+
+				// Update the InstrumentationConfig status
+				err := clientset.Status().Update(
+					ctx,
+					workloadInstrumentationConfigReference,
+				)
+				if err != nil {
+					return fmt.Errorf("failed to update status for daemonset %s in namespace %s: %v", ds.Name, ds.Namespace, err)
+				}
+			}
+		}
+	}
+	return nil
+}
+func handleContainerRuntimeDetailsUpdate(
+	containerObject v1.Container,
+	originalWorkloadEnvVar envoverwrite.OrigWorkloadEnvValues,
+	runtimeDetailsByContainer *[]v1alpha1.RuntimeDetailsByContainer,
+) error {
+	for i := range *runtimeDetailsByContainer {
+		containerRuntimeDetails := &(*runtimeDetailsByContainer)[i]
+
+		// Find the relevant container in runtimeDetailsByContainer
+		if containerRuntimeDetails.ContainerName != containerObject.Name {
+			continue
+		}
+
+		// Process environment variables for the container
+		annotationEnvVarsForContainer := originalWorkloadEnvVar.GetContainerStoredEnvs(containerObject.Name)
+		for envKey, envValue := range annotationEnvVarsForContainer {
+			isEnvVarAlreadyExists := isEnvVarPresent(containerRuntimeDetails.EnvFromContainerRuntime, envKey)
+			// The runtimeDetails might already contain the environment variable if Odiglet started before the migration was executed and modified the env vars.
+			// In this case, we want to overwrite the value set by Odiglet with a new one.
+			if isEnvVarAlreadyExists {
+				containerRuntimeDetails.EnvFromContainerRuntime = removeEnvVar(containerRuntimeDetails.EnvFromContainerRuntime, envKey)
+			}
+
+			// Handle runtime-originated environment variables
+			if envValue == nil {
```

Comment:
I might be missing something.
If an env var originated from a Dockerfile, it won't be part of the annotation map, right?
If that is the case, the key will not be part of the map.
This code assumes that env vars that are not from the k8s manifest will appear in the annotation with nil value - is this the case?

Reply:
```
odigos.io/manifest-env-original-val: '{"simple-app":{"PYTHONPATH":null}}'
odigos.io/manifest-env-original-val: '{"coupon":{"NODE_OPTIONS":null}}' 
```